src/App.js

```diff
@@ -1,25 +1,78 @@
 import React from 'react';
+import Clock from './Clock';
 
 import './App.scss';
 
-const App = () => {
-  setInterval(() => {
-    const date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
-};
+class App extends React.Component {
+  state = {
+    clockName: 'Clock: 0',
+    isClockVisible: true,
+  };
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevState.clockName !== this.state.clockName) {
+      window.console.log(
+        // eslint-disable-next-line
+        `The Clock was renamed from ${prevState.clockName} to ${this.state.clockName}.`,
+      );
+    }
+  }
+
+  setRandomName() {
+    this.setState({
+      clockName: `Clock name: ${Math.floor(Math.random() * 100)}`,
+    });
+  }
+
+  showClock() {
+    this.setState({
+      isClockVisible: true,
+    });
+  }
+
+  hideClock() {
+    this.setState({
+      isClockVisible: false,
+    });
+  }
+
+  render() {
+    const { isClockVisible, clockName } = this.state;
+
+    return (
+      <>
+        <div className="App">
+          <h1>React clock</h1>
+          <p>
+            Current time:
+            {' '}
+            { isClockVisible && <Clock name={clockName} />}
+          </p>
+        </div>
+        <button
+          type="button"
+          className="showClock"
+          onClick={this.showClock.bind(this)}
```

Comment:
```onClick={this.showClock}```

Reply:
Nice that you've used `bind`! 👍 🚀 
Though it's not necessary here, you could just use `hideClock = () => {`
But ok